### PR TITLE
refactor: remove usage of assert from cache.js [#973]

### PR DESCRIPTION
**Pull request by awshanks**
Type:

The following has been addressed in the PR:

remove all import assert from 'assert'; in favor of Jest matchers in cache.js

### CODE REVIEW

File: test/functional/uplinks/cache.js [hunk 1]
```diff
@@ -35,7 +34,7 @@ export default function (server, server2, server3) {
 
     beforeAll(function () {
       const pkg = require('../fixtures/package')(PKG_GH131);
-      pkg.dist.shasum = createTarballHash().update(getBinary()).digest('hex');
+      pkg.dist.shasum = crypto.createHash('sha1').update(getBinary()).digest('hex');
```
[the remark this answers is not in the document]
Comment (awshanks) [reply test/functional/uplinks/cache.js]: I may have accidentally brought over the version from master, did not intend for that change.

[the remark this answers is not in the document]
Comment (awshanks) [reply test/functional/uplinks/cache.js]: I have resolved the above with the latest commit.

**Comment by awshanks** [comment]
@juanpicado can you kick off the test_node8 again in CircleCI.  Looks to be a test unrelated to the changes made

File: test/functional/uplinks/cache.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (juanpicado) [reply test/functional/uplinks/cache.js]: No problem :) Perfect

File: test/unit/api/no_proxy.spec.js [hunk 2]
```diff
@@ -11,78 +10,78 @@ function setupProxy(host, config, mainconfig) {
...
+    expect(x.proxy).toEqual('123');
     x = setupProxy('http://x/x', {http_proxy: '123', no_proxy: ''}, {});
-    assert.equal(x.proxy, '123');
+    expect(x.proxy).toEqual('123');
   });
 
   test('no_proxy - simple/include', () => {
     let x = setupProxy('http://localhost', {http_proxy: '123'}, {no_proxy: 'localhost'});
-    assert.equal(x.proxy, undefined);
+    expect(x.proxy).toEqual(undefined);
```
Comment (juanpicado) [inline test/unit/api/no_proxy.spec.js]: Here also is valid ` expect(x.proxy).not.toBeDefined();`

**Pull request merged**